### PR TITLE
Update config doc.

### LINKED ISSUE #1377: Document config.toml
In GitLab by @damip

Write a documentation for all config.toml entries, both in node and in client

### PR DESCRIPTION
**Pull request by AureliaDolo**
Fix #1377

### CODE REVIEW

**Comment by AureliaDolo** [comment]
@Noxhaa If you wanna take a look, so you might help people adjust their config to maybe have better stability

**Comment by Noxhaa** [comment]
Nice comments, it will help for sure !

File: massa-node/base_config/config.toml [hunk 1]
```diff
@@ -1,96 +1,175 @@
 
 
 [logging]
+    # increase for more info about the node. May reduce your node ability to keep up with the network
```
Comment (damip) [inline massa-node/base_config/config.toml:4]: Logging level. High log levels might impact performance. 0: ERROR, 1: WARN, 2: INFO, 3: DEBUG, 4: TRACE

File: massa-node/base_config/config.toml [hunk 2]
```diff
@@ -1,96 +1,175 @@
...
     level = 2
 
 [api]
+    # max number of future periods considered during requests
     draw_lookahead_period_count = 10
+    # where to send admin level requests (node management)
```
Comment (damip) [inline massa-node/base_config/config.toml:10]: address:port on which the node API listens for admin and node management requests. Dangerous if publicly exposed.

File: massa-node/base_config/config.toml [hunk 3]
```diff
@@ -1,96 +1,175 @@
...
     bind_private = "127.0.0.1:33034"
+    # where to send public requests (get data, send operation)
```
Comment (damip) [inline massa-node/base_config/config.toml:12]: address:port on which the node API listens for public requests. Can be exposed to the Internet.

File: massa-node/base_config/config.toml [hunk 4]
```diff
@@ -1,96 +1,175 @@
...
     bind_public = "0.0.0.0:33035"
+    # max number items accepted
```
Comment (damip) [inline massa-node/base_config/config.toml:14]: max number of arguments per RPC call

File: massa-node/base_config/config.toml [hunk 5]
```diff
@@ -1,96 +1,175 @@
...
     max_arguments = 128
 
 [execution]
+    # path to the initial sequential balance ledger
```
Comment (damip) [inline massa-node/base_config/config.toml:18]: path to the initial smart contract balance ledger

File: massa-node/base_config/config.toml [hunk 6]
```diff
@@ -1,96 +1,175 @@
...
     initial_sce_ledger_path = "base_config/initial_sce_ledger.json"
 
 [consensus]
+    # max number of previously discarded blocks kept in RAM
     max_discarded_blocks = 100
+    # if a block  is future_block_processing_max_periods periods in the future, it is just discarded
```
Comment (damip) [inline massa-node/base_config/config.toml:24]: ...at least future_block_processing_max_periods in the future...

File: massa-node/base_config/config.toml [hunk 7]
```diff
@@ -1,96 +1,175 @@
...
     future_block_processing_max_periods = 100
+    # max number of blocks in the future kept in RAM
     max_future_processing_blocks = 400
+    # max number of blocks waiting for dependencies
     max_dependency_blocks = 2048
+    # number of final periods that must be kept at all time (increse to more resilience to short network disconnections)
```
Comment (damip) [inline massa-node/base_config/config.toml:30]: at all times

increase for more resilience

. High values will increase RAM usage.

File: massa-node/base_config/config.toml [hunk 8]
```diff
@@ -1,96 +1,175 @@
...
+    force_keep_final_periods = 20
+
+    # max number of time the node tries to fill up a block before producing it
```
Comment (damip) [inline massa-node/base_config/config.toml:33]: max number of batches of pool operations to scan when generating a block

File: massa-node/base_config/config.toml [hunk 9]
```diff
@@ -1,96 +1,175 @@
...
     max_operations_fill_attempts = 102400
+    # number of operation to include in a block
```
Comment (damip) [inline massa-node/base_config/config.toml:35]: size of a pool operation batch for block creation

File: massa-node/base_config/config.toml [hunk 10]
```diff
@@ -1,96 +1,175 @@
...
+    operation_batch_size = 1024
+
+    # path to the cold sored ledger
```
Comment (damip) [inline massa-node/base_config/config.toml:38]: path to the ledger file

File: massa-node/base_config/config.toml [hunk 11]
```diff
@@ -1,96 +1,175 @@
...
     ledger_path = "storage/ledger"
+    # size of the ledger kept in cache
     ledger_cache_capacity = 1048576
+    # [optionnal] the ledger is saved on disk every ledger_flush_interval
     ledger_flush_interval = 10000
+    # wether to overwrite the ledger at start
```
Comment (damip) [inline massa-node/base_config/config.toml:44]: whether

File: massa-node/base_config/config.toml [hunk 12]
```diff
@@ -1,96 +1,175 @@
...
     ledger_reset_at_startup = true
+    # path to the initial ledger
```
Comment (damip) [inline massa-node/base_config/config.toml:46]: path to the initial consensus balance ledger

File: massa-node/base_config/config.toml [hunk 13]
```diff
@@ -1,96 +1,175 @@
...
+
+    # max millis to wait while sending an event before dropping it
     max_send_wait = 500
-    force_keep_final_periods = 20
+    # useless blocks are pruned every block_db_prune_interval ms
+    block_db_prune_interval = 5000
+
+    # path to the initial roll distribution
     initial_rolls_path = "base_config/initial_rolls.json"
+    # number of cached cycle for proof of stake seed initialization
```
Comment (damip) [inline massa-node/base_config/config.toml:56]: number of cycles cached by the proof-of-stake selection system

File: massa-node/base_config/config.toml [hunk 14]
```diff
@@ -1,96 +1,175 @@
...
-    block_db_prune_interval = 5000
+    # max number of item returned per query
     max_item_return_count = 100
 
 [protocol]
+    # timeout after which without answer a hanshake is ended
     message_timeout = 5000
+    # timeout after whick we consider a node does not have the block we asked for
     ask_block_timeout = 10000
+    # max number for the local vision of what blocks a node knows
```
Comment (damip) [inline massa-node/base_config/config.toml:71]: max cache size for which blocks a foreign node knows about

File: massa-node/base_config/config.toml [hunk 15]
```diff
@@ -1,96 +1,175 @@
...
     max_node_known_blocks_size = 1024
+    # max number for the local vision of what blocks a node wants
```
Comment (damip) [inline massa-node/base_config/config.toml:73]: max cache size for which blocks a foreign node asked for

File: massa-node/base_config/config.toml [hunk 16]
```diff
@@ -1,96 +1,175 @@
...
     max_node_wanted_blocks_size = 1024
+    # max number of blocks we can asked per node
```
Comment (damip) [inline massa-node/base_config/config.toml:75]: asked => ask simultaneously

File: massa-node/base_config/config.toml [hunk 17]
```diff
@@ -1,96 +1,175 @@
...
     max_simultaneous_ask_blocks_per_node = 2048
+    # max millis to wait while sending an event before dropping it
     max_send_wait = 500
+    # max number for the local vision of what operations a node knows
```
Comment (damip) [inline massa-node/base_config/config.toml:79]: max cache size for which operations a foreign node knows about

File: massa-node/base_config/config.toml [hunk 18]
```diff
@@ -1,96 +1,175 @@
...
     max_known_ops_size = 10024
+    # max number for the local vision of what endorsements a node knows about
```
Comment (damip) [inline massa-node/base_config/config.toml:81]: max cache size for which endorsements a foreign node knows about

File: massa-node/base_config/config.toml [hunk 19]
```diff
@@ -1,96 +1,175 @@
...
     max_known_endorsements_size = 1024
 
 [network]
+    # where to listen for communication
```
Comment (damip) [inline massa-node/base_config/config.toml:85]: address:port on which to listen for protocol communication

File: massa-node/base_config/config.toml [hunk 20]
```diff
@@ -1,96 +1,175 @@
...
     bind = "[::]:31244"
+    # port used by protocol
     protocol_port = 31244
+    # millis spent waiting for a respone
```
Comment (damip) [inline massa-node/base_config/config.toml:89]: timeout for connection establishment

File: massa-node/base_config/config.toml [hunk 21]
```diff
@@ -1,96 +1,175 @@
...
     connect_timeout = 3000
+    # connection attempt to avaible peers every wakeup_interval millis
```
Comment (damip) [inline massa-node/base_config/config.toml:91]: attempt a connection to available peers when needed every wakeup_interval milliseconds

File: massa-node/base_config/config.toml [hunk 22]
```diff
@@ -1,96 +1,175 @@
...
     wakeup_interval = 5000
+    # path to the cold stored peers
```
Comment (damip) [inline massa-node/base_config/config.toml:93]: path to the local peers storage file

File: massa-node/base_config/config.toml [hunk 23]
```diff
@@ -1,96 +1,175 @@
...
     peers_file = "storage/peers.json"
+    # path to the initial peer file
```
Comment (damip) [inline massa-node/base_config/config.toml:95]: peers

File: massa-node/base_config/config.toml [hunk 24]
```diff
@@ -1,96 +1,175 @@
...
     initial_peers_file = "base_config/initial_peers.json"
+    # target number of connection to a bootstrap node, once the bootstrap succeeded
```
Comment (damip) [inline massa-node/base_config/config.toml:97]: number of connections to bootstrap nodes

File: massa-node/base_config/config.toml [hunk 25]
```diff
@@ -1,96 +1,175 @@
...
     target_bootstrap_connections = 1
+    # max number of connection attempts to bootstrap nodes, once the bootstrap succeeded
```
Comment (damip) [inline massa-node/base_config/config.toml:99]: max number of simultaneous protocol connection attempts towards bootstrap nodes

File: massa-node/base_config/config.toml [hunk 26]
```diff
@@ -1,96 +1,175 @@
...
     max_out_bootstrap_connection_attempts = 1
+    # target number of non bootstrap out connections
```
Comment (damip) [inline massa-node/base_config/config.toml:101]: out => outgoing

File: massa-node/base_config/config.toml [hunk 27]
```diff
@@ -1,96 +1,175 @@
...
     target_out_nonbootstrap_connections = 10
+    # max number of in non bootstrap connections
```
Comment (damip) [inline massa-node/base_config/config.toml:103]: in => inbound

File: massa-node/base_config/config.toml [hunk 28]
```diff
@@ -1,96 +1,175 @@
...
     max_in_nonbootstrap_connections = 15
+    # max number of in connections per ip
```
Comment (damip) [inline massa-node/base_config/config.toml:105]: in => inbound

File: massa-node/base_config/config.toml [hunk 29]
```diff
@@ -1,96 +1,175 @@
...
     max_in_connections_per_ip = 5
+    # max number of out non bootstrap connection attempts
```
Comment (damip) [inline massa-node/base_config/config.toml:107]: out => outgoing

simultaneous

File: massa-node/base_config/config.toml [hunk 30]
```diff
@@ -1,96 +1,175 @@
...
     max_out_nonbootstrap_connection_attempts = 10
+    # max number of stored idle peers
     max_idle_peers = 10000
+    # max number of stored banned peers
     max_banned_peers = 100
+    # max number of advertized peers
     max_advertise_length = 10000
+    # peers are cold stored every peers_file_dump_interval millis
```
Comment (damip) [inline massa-node/base_config/config.toml:115]: cold stored => dumped to file

File: massa-node/base_config/config.toml [hunk 31]
```diff
@@ -1,96 +1,175 @@
...
     peers_file_dump_interval = 10000
+    # max size of sent messages
     max_message_size = 1048576000
+    # after message_timeout millis we are no longer waiting on handshake message
```
Comment (damip) [inline massa-node/base_config/config.toml:119]: timeout when waiting for a message from a foreign node

File: massa-node/base_config/config.toml [hunk 32]
```diff
@@ -1,96 +1,175 @@
...
     message_timeout = 5000
+    # Every ask_peer_list_interval in millis we ask every one for its advertisable peers list.
```
Comment (damip) [inline massa-node/base_config/config.toml:121]: interval in milliseconds for asking peer lists from peers we are connected to

File: massa-node/base_config/config.toml [hunk 33]
```diff
@@ -1,96 +1,175 @@
...
     ask_peer_list_interval = 30000
+    # path to the node private key (not the staking key)
     private_key_file = "config/node_privkey.key"
+    # max number of asked blocks per message
     max_ask_blocks_per_message = 128
+    # max number of operation per message
```
Comment (damip) [inline massa-node/base_config/config.toml:127]: operations

File: massa-node/base_config/config.toml [hunk 34]
```diff
@@ -1,96 +1,175 @@
...
     max_operations_per_message = 1024
+    # max number of endorsements per message
     max_endorsements_per_message = 1024
+    # max millis to wait while sending an event before dropping it
     max_send_wait = 500
+    # we forget we banned a node after ban_timeout
```
Comment (damip) [inline massa-node/base_config/config.toml:133]: milliseconds

File: massa-node/base_config/config.toml [hunk 35]
```diff
@@ -1,96 +1,175 @@
...
     ban_timeout = 3600000
 
 [bootstrap]
+    # list of bootstrap (ip, public key)
```
Comment (damip) [inline massa-node/base_config/config.toml:137]: public key => node ID

File: massa-node/base_config/config.toml [hunk 36]
```diff
@@ -1,96 +1,175 @@
...
     bootstrap_list = [
         ["149.202.86.103:31245", "5GcSNukkKePWpNSjx9STyoEZniJAN4U4EUzdsQyqhuP3WYf6nj"],
         ["149.202.89.125:31245", "5wDwi2GYPniGLzpDfKjXJrmHV3p1rLRmm4bQ9TUWNVkpYmd4Zm"],
         ["158.69.120.215:31245", "5QbsTjSoKzYc8uBbwPCap392CoMQfZ2jviyq492LZPpijctb9c"],
         ["158.69.23.120:31245", "8139kbee951YJdwK99odM7e6V3eW7XShCfX5E2ovG3b9qxqqrq"]
     ]
+    # [optionnal] for bootstrap servers to port people will bootstrap from
```
Comment (damip) [inline massa-node/base_config/config.toml:144]: [optional] address:port on which to listen for incoming bootstrap requests

File: massa-node/base_config/config.toml [hunk 37]
```diff
@@ -1,96 +1,175 @@
...
     bind = "[::]:31245"
+    # timeout to establish a bootstrap connection
     connect_timeout = 15000
+    # delay before an other bootsrap attempt
```
Comment (damip) [inline massa-node/base_config/config.toml:148]: delay in milliseconds to wait between consecutive bootstrap attempts

File: massa-node/base_config/config.toml [hunk 38]
```diff
@@ -1,96 +1,175 @@
...
     retry_delay = 5000
+    # if ping is too high bootstrap will be inturrupted
```
Comment (damip) [inline massa-node/base_config/config.toml:150]: milliseconds

File: massa-node/base_config/config.toml [hunk 39]
```diff
@@ -1,96 +1,175 @@
...
     max_ping = 10000
+    # after read timeout millis, we stop reading
```
Comment (damip) [inline massa-node/base_config/config.toml:152]: timeout for incoming message readout

File: massa-node/base_config/config.toml [hunk 40]
```diff
@@ -1,96 +1,175 @@
...
     read_timeout = 10000
+    # after write_timeout millis, we stop writing
```
Comment (damip) [inline massa-node/base_config/config.toml:154]: timeout for message sending

File: massa-node/base_config/config.toml [hunk 41]
```diff
@@ -1,96 +1,175 @@
...
     write_timeout = 10000
+    # if enabled, time will be synchronized on the server time
```
Comment (damip) [inline massa-node/base_config/config.toml:156]: when enabled, apply a correction to the local computer clock to match bootstrap server time

File: massa-node/base_config/config.toml [hunk 42]
```diff
@@ -1,96 +1,175 @@
...
     enable_clock_synchronization = false
+    # [server] data is kept for cache duration millis
```
Comment (damip) [inline massa-node/base_config/config.toml:158]: bootstrap data is cached for ...

File: massa-node/base_config/config.toml [hunk 43]
```diff
@@ -1,96 +1,175 @@
...
     cache_duration = 15000
+    # max number of simulataneous bootstraps for server
     max_simultaneous_bootstraps = 2
+    # max size of recently bootstraped ips
```
Comment (damip) [inline massa-node/base_config/config.toml:162]: max size of recently bootstrapped IP cache

File: massa-node/base_config/config.toml [hunk 44]
```diff
@@ -1,96 +1,175 @@
...
     ip_list_max_size = 10000
+    # after a botstrap attemp, a peer must wait per_ip_min_interval millis
```
Comment (damip) [inline massa-node/base_config/config.toml:164]: refuse consecutive bootstrap attempts from a given IP when the interval between them is lower than per_ip_min_interval milliseconds

File: massa-node/base_config/config.toml [hunk 45]
```diff
@@ -1,96 +1,175 @@
...
     per_ip_min_interval = 300000
 
 [pool]
+    # max number of operation kept per thread
```
Comment (damip) [inline massa-node/base_config/config.toml:168]: operations

File: massa-node/base_config/config.toml [hunk 46]
```diff
@@ -1,96 +1,175 @@
...
     max_pool_size_per_thread = 100000
+    # if an operation is too much in the future it will be ignored
     max_operation_future_validity_start_periods = 100
+    # max number of endorsement kept
```
Comment (damip) [inline massa-node/base_config/config.toml:172]: endorsements

File: massa-node/base_config/config.toml [hunk 47]
```diff
@@ -1,96 +1,175 @@
...
     max_endorsement_count = 10000
+    # max number of item returned per query
```
Comment (damip) [inline massa-node/base_config/config.toml:174]: items

**Review by damip — CHANGES_REQUESTED** [review]

**Review by damip — CHANGES_REQUESTED** [review]
just a couple more

**Review by damip — APPROVED** [review]

**Comment by AureliaDolo** [comment]
bors merge

**Pull request merged**